vllm/_custom_ops.py

```diff
@@ -1104,14 +1128,11 @@ def scaled_fp4_experts_quant(
     assert input_tensor.ndim == 2, (
         f'input.ndim needs to be == 2, but got {input_tensor.ndim}.')
 
-    input_tensor = input_tensor[
-        expert_map] if expert_map is not None else input_tensor
-    m_numtopk, k = input_tensor.shape
-    # Control the maximum number of tokens per expert supported by the
-    # NVFP4 MoE Expert Quantization. This is used to prevent the kernel
-    # from running out of memory. This value can also be increased to support
-    # larger models.
     MAX_TOKENS_PER_EXPERT = envs.VLLM_MAX_TOKENS_PER_EXPERT_FP4_MOE
+    input_tensor = shuffle_rows(
+        input_tensor, shuffle_map) if shuffle_map is not None else input_tensor
```

Comment:
Why not just call `shuffle_rows` in `cutlass_moe_fp4` before calling `scaled_fp4_experts_quant`? Then we can remove shuffle_map from this function.
If there is no functional or perf difference, I think it is more clear to take it out of this function like `c2 = ops.shuffle_rows(c2, c_map)` already is called after the last `cutlass_fp4_moe_mm`